cpp/src/arrow/memory_pool.cc

```diff
@@ -253,6 +253,8 @@ class SystemAllocator {
 #endif
     }
   }
+
+  static void ReleaseUnused() {}
```

Comment:
Note the glibc has [malloc_trim](https://man7.org/linux/man-pages/man3/malloc_trim.3.html). This can probably be used depending if the `__GLIBC__` macro is present.